### PR TITLE
Feature/5 get users

### LINKED ISSUE #5: Get users - happy path
### AC

- Add endpoint to find user by id
- Add end point to find all users paginated and with following filters
  - add filter parameter 'query', which will search whether the text is included in name or email
  - add filter parameter 'status'. If provided, it will return based on status. otherwise all
- Add Integration Tests
- Add Unit Tests
- Understanding of QueryDsl and Specification

### References
- https://reflectoring.io/spring-boot-paging/
- https://howtodoinjava.com/spring-boot2/testing/spring-integration-testing/
- https://rieckpil.de/guide-to-springboottest-for-spring-boot-integration-tests/
- https://reflectoring.io/spring-boot-test/

- https://www.softwaretestingclass.com/unit-testing-test-case-preparation-guidelines/
- https://www.testim.io/blog/unit-testing-best-practices/
- https://www.partech.nl/nl/publicaties/2020/03/10-unit-testing-best-practices#

### PR DESCRIPTION
**Pull request by jyotikakrani25**
Added integration test as part of this PR.

### CODE REVIEW

File: src/main/java/com/buddy/tutorial/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@AllArgsConstructor
+public class UserController {
+
+    @Autowired
```
Comment (habelMarottipuzha) [inline src/main/java/com/buddy/tutorial/controller/UserController.java]: whats the need for field based and constructor based injection. use any one

File: src/main/java/com/buddy/tutorial/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private final UserService userService;
+
+    /**
+     * This API will create users in database.
+     *
+     * @param user is considering as a Request Body
+     * @return user.
+     */
+    @PostMapping("/users")
```
Comment (habelMarottipuzha) [inline src/main/java/com/buddy/tutorial/controller/UserController.java]: move common path of all controllers to class level

File: src/main/java/com/buddy/tutorial/service/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public User getUserById(Integer userId) {
+        return userRepository.findById(userId).orElseThrow(() -> new RuntimeException("User Not Found"));
+    }
+
+    @Override
+    public Page<User> getAllUsers(Pageable pageable, String query, UserStatus status) {
+        if (query != null && status != null) {
+            return userRepository.findByNameContainingIgnoreCaseOrEmailContainingIgnoreCaseAndStatus(query, query, status, pageable);
+        } else if (query != null) {//select U* from users where name = %s and email = %s : select * from Users where name = buddy ,or email =
```
Comment (habelMarottipuzha) [inline src/main/java/com/buddy/tutorial/service/UserServiceImpl.java]: remove comment

File: README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Spring Boot tutorial
```
Comment (habelMarottipuzha) [inline README.md:1]: update readme. include more information regarding project. also include various steps to run check style, build etc..

**Review by habelMarottipuzha — CHANGES_REQUESTED** [review]

File: src/test/java/com/buddy/tutorial/controller/UserControllerIT.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
```
Comment (habelMarottipuzha) [inline src/test/java/com/buddy/tutorial/controller/UserControllerIT.java:22]: use sliced tests here. https://developer.okta.com/blog/2021/07/12/spring-boot-test-slices     https://reflectoring.io/spring-boot-test/

File: src/main/java/com/buddy/tutorial/service/UserServiceImpl.java (hunk 3 above)
Comment (jyotikakrani25) [reply src/main/java/com/buddy/tutorial/service/UserServiceImpl.java]: Done

File: src/test/java/com/buddy/tutorial/TutorialApplicationTests.java [hunk 6]
```diff
@@ -8,6 +9,8 @@ class TutorialApplicationTests {
 
 	@Test
 	void contextLoads() {
-	}
 
+        Assertions.assertTrue(true);
```
Comment (habelMarottipuzha) [inline src/test/java/com/buddy/tutorial/TutorialApplicationTests.java]: Whas this?? add some test to test context

**Review by habelMarottipuzha — APPROVED** [review]

**Pull request merged**